### ISSUE TITLE
BitMEX private_get_position incorrect signature in Python/asyncio

### ISSUE THREAD

**Issue by andersea** [ccxt/ccxt#661]
Hi. I am seeing something a bit odd. The async version of the exchange specific api call `private_get_position` fails with this error:
```
ccxt.base.errors.AuthenticationError, bitmex GET https://testnet.bitmex.com/api/v1/position?filter=%7B%22isOpen%22%3A+true%2C+%22symbol%22%3A+%22XBTUSD%22%7D&columns=homeNotional%2CforeignNotional%2CmarkPrice 401 {"error":{"message":"Signature not valid.","name":"HTTPError"}}
```
The sync version works. 

Here is some test code:

```python
"""Application entry point."""
import asyncio
import json
import ccxt
import ccxt.async as ccxt_async
import config


def test(symbol):
    """Position api test function."""
    bitmex = ccxt.bitmex({
        **config.API_KEYS['bitmex'],
        'nonce': ccxt.Exchange.milliseconds
    })

    bitmex.verbose = True
    bitmex.urls['api'] = bitmex.urls['test']

    return bitmex.private_get_position({
        'filter': json.dumps({
            "isOpen": True,
            "symbol": symbol
        }),
        'columns': 'homeNotional,foreignNotional,markPrice'
    })

async def test_async(symbol):
    """Position api test function."""
    bitmex = ccxt_async.bitmex({
        **config.API_KEYS['bitmex'],
        'nonce': ccxt.Exchange.milliseconds
    })

    bitmex.verbose = True
    bitmex.urls['api'] = bitmex.urls['test']

    return await bitmex.private_get_position({
        'filter': json.dumps({
            "isOpen": True,
            "symbol": symbol
        }),
        'columns': 'homeNotional,foreignNotional,markPrice'
    })


if __name__ == '__main__':
    print(test('XBTUSD'))
    asyncio.get_event_loop().run_until_complete(test_async('XBTUSD'))
```
This code expects a config.py with bitmex testnet api keys:
```
API_KEYS = {
    'bitmex': {
        'apiKey': 'key_id',
        'secret': 'key_secret'
    }
}
```
Also there should probably be an open BTXUSD position on the account to fetch.

Environment:

* Python 3.6.3 (default, Oct  3 2017, 21:45:48)
* ccxt 1.10.196

**Comment by kroitor** [comment]
@andersea thanks for this comprehensive report! I will look into it and will fix it asap.

**Comment by kroitor** [comment]
@andersea 

Does this work

```
"""Application entry point."""
import asyncio
import json
import ccxt.async as ccxt_async
import config

async def test_async(symbol):
    """Position api test function."""
    bitmex = ccxt_async.bitmex({
        **config.API_KEYS['bitmex'],
        'nonce': ccxt.Exchange.milliseconds
    })

    bitmex.verbose = True
    bitmex.urls['api'] = bitmex.urls['test']

    return await bitmex.private_get_position({
        'filter': json.dumps({
            "isOpen": True,
            "symbol": symbol
        }),
        'columns': 'homeNotional,foreignNotional,markPrice'
    })


if __name__ == '__main__':
    asyncio.get_event_loop().run_until_complete(test_async('XBTUSD'))
```

?

If it does work without the sync call, then it indicates that you are running into a nonce conflict, when you try to fetch a private call with a sync API, then with async API immediately after that.

More on this: https://github.com/ccxt/ccxt/issues/147

**Comment by kroitor** [comment]
This is related to filter encoding, because if we comment those params, it works in both cases, while having them uncommented, both sync and async fail on our side. We are investigating into it and will fix it soon, hopefully.

**Comment by andersea** [comment]
Thanks! I appreciate the update! That is really odd though. On my side the sync side works fine: Here is the output (I disabled verbose).

```
[{'account': 10083, 'symbol': 'XBTUSD', 'currency': 'XBt', 'timestamp': '2017-11-25T11:08:45.107Z', 'currentQty': -299, 'simpleQty': -0.0402, 'markPrice': 8383, 'liquidationPrice': 100000000, 'homeNotional': -0.03566771, 'foreignNotional': 299}]
ccxt.base.errors.AuthenticationError, bitmex GET https://testnet.bitmex.com/api/v1/position?filter=%7B%22isOpen%22%3A+true%2C+%22symbol%22%3A+%22XBTUSD%22%7D&columns=homeNotional%2CforeignNotional%2CmarkPrice 401 {"error":{"message":"Signature not valid.","name":"HTTPError"}}
```

**Comment by andersea** [comment]
Woops reopen

**Comment by kroitor** [comment]
@andersea can you please update to most recent version of ccxt and post verbose output for both sync/async as seen on your side? This will help us finally close this issue.

**Comment by kroitor** [comment]
The problem is most likely not related to signatures, but it is related to simultaneous usage of the same API key with conflicting nonces.

**Comment by kroitor** [comment]
@andersea also, can you run two separate tests to confirm they don't intersect in nonce numbers?

**Comment by andersea** [comment]
Here is my current test code:

https://gist.github.com/andersea/55821130219b4eb8f222f4851867aa8b

And here is the output.

https://gist.github.com/andersea/cc7df227c6b6ea4229f8b97ae89732de

Looks like the nonce is increasing. I also tried commenting out the all synchronous code, including the synchronous test funcing and only running the async code. The async test code fails every time for me.

Here is the async only version:

https://gist.github.com/andersea/72cd17bca89d3a618519f72e81ea851d

**Comment by kovalevvlad** [comment]
Same issue with `fetch_open_positions`. I'll see if I can figure it out.

**Comment by kovalevvlad** [comment]
Okay, this is happening due to this issue in aiohttp - https://github.com/aio-libs/aiohttp/issues/2188.
I'll sort it out.

**Comment by kroitor** [comment]
@kovalevvlad thanks! this issue has been bugging me for long...

**Comment by kroitor** [comment]
Fixed in #1905, ccxt version 1.10.1107. Thx to @kovalevvlad

**Comment by andersea** [comment]
Thank you so much. Great work!
